app/src/main/java/com/example/tasky/feature_agenda/presentation/datepicker/MonthPickerOnToolbar.kt

```diff
@@ -0,0 +1,229 @@
+    MonthTextWithIcon(
+        monthSelected = "MARCH",
+        onClick = { }
+    )
+}
+
+@Composable
+fun MonthPickerOnToolbar(
+    monthSelected: String,
+    onMonthSelected: (String) -> Unit,
+    dialogState: MaterialDialogState,
+    onDialogStateChange: (MaterialDialogState) -> Unit
+) {
+
+    // Button to open the DatePickerDialog
+    MonthTextWithIcon(
+        monthSelected = monthSelected,
+        onClick = {
+            dialogState.show()
+            onDialogStateChange(dialogState)
+        }
+    )
+
+    MaterialDialog(
+        dialogState = dialogState,
+        buttons = {
+            positiveButton(
+                stringResource(id = R.string.ok),
+                textStyle = TextStyle(color = Color.Black)
+            )
+            negativeButton(
+                stringResource(id = R.string.cancel),
+                textStyle = TextStyle(color = Color.Black)
+            )
+        }
+    ) {
+        datepicker(
+            yearRange = IntRange(2023, 2030),
+            colors = customDatePickerColors()
+        ) { date ->
+            onMonthSelected(date.month.toString())
+            dialogState.hide()
+            onDialogStateChange(dialogState)
+        }
+    }
+}
+
+@Composable
+fun customDatePickerColors(): com.vanpra.composematerialdialogs.datetime.date.DatePickerColors {
+    return com.vanpra.composematerialdialogs.datetime.date.DatePickerDefaults.colors(
+        headerBackgroundColor = colorResource(id = R.color.tasky_green),
+        headerTextColor = Color.White,
+        calendarHeaderTextColor = Color.Black,
+        dateActiveBackgroundColor = colorResource(id = R.color.sea_foam_green),
+        dateInactiveBackgroundColor = Color.White,
+        dateActiveTextColor = Color.Black,
+        dateInactiveTextColor = Color.Black
+    )
+}
+
+@Composable
+@Preview
+fun CalendarDatePicker() {
+    val dialogState = rememberMaterialDialogState()
+    MaterialDialog(
+        dialogState = dialogState,
+        buttons = {
+            positiveButton(stringResource(id = R.string.ok))
+            negativeButton(stringResource(id = R.string.cancel))
+        }
+    ) {
+        datepicker(
+            yearRange = IntRange(2023, 2030)
+        ) { date ->
+            dialogState.hide()
+        }
+    }
+    /* This should be called in an onClick or an Effect */
+    dialogState.show()
+}
+
+//TODO not used, but maybe in future use Material3 date picker
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun Material3MonthPickerOnToolbar(
+    monthSelected: String,
+    openDatePickerDialog: Boolean,
+    onMonthClick: () -> Unit,
+    onMonthSelected: (String) -> Unit
+) {
+    // Initial state setup for the DatePickerDialog. Specifies to show the picker initially
+    val datePickerState = rememberDatePickerState(initialDisplayMode = DisplayMode.Picker)
+    // State to hold the selected date as a String
+    val selectedDateLabel = remember { mutableStateOf(monthSelected) }
```

Comment:
Does the comment really add any value? In most cases, if you need a comment to explain **what** happens, the code itself is not readable enough (or it's just unnecessary like here, since anyone will know what `selectedDateLabel` represents). Comments are great to explain **why** something happens.

Same counts for all other comments in this composable.

Reply:
haha the dangers of copy/paste.. I will clean this  [material3 DatePicker](https://medium.com/deuk/implementing-material-design-3-datepicker-in-android-compose-96d25cf0a468)